openid-4-verifiable-presentations-1_0.md

```diff
@@ -228,6 +275,84 @@ OpenID for Verifiable Presentations extends existing OAuth 2.0 mechanisms as fol
+
+The Discovery Request message has the following parameters:
+
+`presentation_request_uri`:
+: A string containing an HTTPS URL pointing to a resource under the control of the Verifier where the Wallet is supposed to obtain the presentation request object. 
+
+`context`:
+: A string identifying the context of the discovery request from the perspective of the Verifier. The value is opqaue to the Wallet, it MUST pass this value to the Presentation Endpoint Request (see below). 
+
+The Discovery Request is represented as a JSON object, where all parameters are top level JSON claims. 
+
+The following is an example request: 
+
+```JSON
+{
+    "presentation_request_uri": "verifier.example.com/presentation_request",
+    "context": "register_kyc"
+}
+```
+
+The Discovery Request is either send the Discovery Endpoint of the Wallet or it is rendered as a QR Code and scanned with the Wallet or a camera app. 
+
+Note: the Discovery Request intentionally does not use OAuth parameters, such as a `client_id`. The idea is to allow the Verifier to select the Client Identifier it wants to use for interacting with the Wallet after it has determined the trust mechanisms and domains the wallet supports.   
+
+## Discovery Endpoint
+
+This is an HTTPS endpoint offered by the Wallet. The Discovery Request MUST be sent as a HTTPS GET request or an HTTP redirect to the Discovery Endpoint URL defined in (#wallet-metadata)
+
+The HTTPS URL contains a single URI query parameter `discovery` where the value is the JSON object as defined above. 
+
+## Presentation Endpoint
+
+The Presentation Endpoint is an HTTPS endpoint exposed by the Verifier. 
+
+### Presentation Request
+
+Presentation Requests MUST be HTTPS POST requests with the "application/json" media type.
+
+The follwowing parameters are defined: 
+
+`context`:
+: A JSON String containing the value of the corresponding Discovery Request's `context` parameter.
+
+`issuer`:
+: A JSON containg a HTTPS URL designating the Issuer URL of the Wallet. 
+
+`w_nonce`:
+: A JSON String containg a cryptographic nonce the Verifier MUST use when creating the signed presentation request object. 
+
+`client_assertion`
+: A JSON String containing a wallet attestation along with a proof of posession of the confiration key as defined in [@!I-D.ietf-oauth-attestation-based-client-auth]. This assertion is used to authenticate the Wallet towards the Verifier. 
+
+### Presentation Response
+
+Presentation Response MUST be HTTPS POST response with the "application/json" media type.
+
+The follwowing parameters are defined: 
+
+`request`:
+: A JSON String containing the signed request object as defined in [@RFC9101]. It MUST fulfill the requirements as defined in (#vp_token_request).
```

Comment:
I think this is confusing. why is there suddenly a need to define a top level claim? usually the entire object in the response is a request object without any top level claims. the parameter name also conflicts with the `request` parameter defined in rfc 9101 to pass the request by value